### PR TITLE
CI: Add tags to aio VMs, periodically clean up stale aio instances (yoga)

### PR DESCRIPTION
**Pull request by markgoddard**
- CI: Add tags to aio VMs
- CI: Add a workflow to clean up stale instances

### CODE REVIEW

File: .github/workflows/stackhpc-ci-cleanup.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: Install OpenStack client
+        run: |
+          pip install python-openstackclient -c https://opendev.org/openstack/requirements/raw/branch/stable/${{ steps.openstack_release.outputs.openstack_release }}/upper-constraints.txt
+
+      - name: Clean up aio instances over 3 hours old
+        run: |
+          result=0
+          changes_before=$(date -Imin -d -3hours)
+          for status in ACTIVE ERROR SHUTOFF; do
```
Comment (Alex-Welsh) [inline .github/workflows/stackhpc-ci-cleanup.yml]: Might want to add `BUILDING` in case anything gets really stuck

File: .github/workflows/stackhpc-ci-cleanup.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+              for instance in $(openstack server list --tags skc-ci-aio --os-compute-api-version 2.66 --format value --column Name --changes-before $changes_before --status $status); do
```
Comment (Alex-Welsh) [inline .github/workflows/stackhpc-ci-cleanup.yml]: listing by ID might be better, since there's no chance of a conflict then. All the data gets printed anyway since `server show` gets run later on

File: .github/workflows/stackhpc-ci-cleanup.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: Determine OpenStack release
+        id: openstack_release
+        run: |
+          BRANCH=$(awk -F'=' '/defaultbranch/ {print $2}' src/kayobe-config/.gitreview)
+          echo "openstack_release=${BRANCH}" | sed "s|stable/||" >> $GITHUB_OUTPUT
+
+      - name: Install OpenStack client
+        run: |
+          pip install python-openstackclient -c https://opendev.org/openstack/requirements/raw/branch/stable/${{ steps.openstack_release.outputs.openstack_release }}/upper-constraints.txt
```
Comment (Alex-Welsh) [inline .github/workflows/stackhpc-ci-cleanup.yml]: Will stable/yoga stay around? might need a way to use different prefixes, but perhaps that should be another PR

File: .github/workflows/stackhpc-ci-cleanup.yml (hunk 1 above)
Comment (markgoddard) [reply .github/workflows/stackhpc-ci-cleanup.yml]: Done

File: .github/workflows/stackhpc-ci-cleanup.yml (hunk 2 above)
Comment (markgoddard) [reply .github/workflows/stackhpc-ci-cleanup.yml]: Done

File: .github/workflows/stackhpc-ci-cleanup.yml (hunk 3 above)
Comment (markgoddard) [reply .github/workflows/stackhpc-ci-cleanup.yml]: We're actually supposed to use https://releases.openstack.org/constraints/upper/yoga

**Review by Alex-Welsh — APPROVED** [review]
